File: market-svc/src/main/java/com/valyalkin/market/ticker/TickersEntity.java
package com.valyalkin.market.ticker;

import com.valyalkin.market.eod.Currency;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.GenericGenerator;

import java.util.UUID;


@NoArgsConstructor
@Entity
@Table(name = "tickers")
@ToString
@Getter
@Setter
public class TickersEntity {

    @Id
    @GeneratedValue(generator = "UUID")
    private UUID id;

    @Column(name = "symbol", unique = true)
    private String symbol;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Currency currency;

    @Column(nullable = false)
    private String name;

    @Column(name = "has_eod", nullable = false)
    private boolean hasEodPrice;

    @Column(nullable = false)
    private String exchange;

    @Column(nullable = false)
    private String acronym;

    @Column(nullable = false)
    private String mic;

    @Column(nullable = false)
    private String country;

    @Column(name = "country_code", nullable = false)
    private String countryCode;
}

File: portfolio-svc/src/main/kotlin/com/valyalkin/portfolio/data/tickers/TickersEntity.kt
package com.valyalkin.portfolio.data.tickers

import com.valyalkin.portfolio.stocks.transactions.Country
import com.valyalkin.portfolio.stocks.transactions.Currency
import jakarta.persistence.*
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

@Entity
@Table(name = "tickers")
class TickersEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID) val id: UUID = UUID.randomUUID(),
    @Column(name = "symbol", unique = true) val symbol: String = "",
    @Enumerated(EnumType.STRING) val currency: Currency = Currency.USD,
    val name: String = "",
    @Column(name = "has_eod") val hasEodPrice: Boolean = false,
    val exchange: String = "",
    val acronym: String = "",
    val mic: String = "",
    val country: String = "",
    @Enumerated(EnumType.STRING) @Column(name = "country_code") val countryCode: Country = Country.US,
)

interface TickersRepository : JpaRepository<TickersEntity, UUID> {
    fun findBySymbol(symbol: String): TickersEntity?
}
